fix: Set ClamAV package names for Ubuntu dists

monoflo opened the pull request:
## Description

This commit modifies the "Determine ClamAV package names" task to include Ubuntu distributions, preventing an undefined variable error when run on such hosts.

## Error Message

```
TASK [cisagov.clamav : Install ClamAV packages] ********************************
fatal: [ubuntu]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'package_names' is undefined\n\nThe error appears to be in '/home/user/.ansible/roles/cisagov.clamav/tasks/main.yml': line 15, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Install ClamAV packages\n  ^ here\n"}
```

## Playbook Workaround

Without this fix, the following workaround in `playbook.yml` is necessary:

```yml
---
- hosts: ubuntu
  gather_facts: yes

  roles:
    - role: cisagov.clamav
      vars:
        package_names: [ clamav-daemon ]  # here
```

monoflo committed:
fix: Set ClamAV package names for Ubuntu dists

This commit modifies the "Determine ClamAV package names" task to
include Ubuntu distributions, preventing an undefined variable error.

jsf9k committed:
Add Ubuntu Bionic and Xenial to role metadata and molecule scenario

jsf9k committed:
Enable and start freshclam for Ubuntu

jsf9k committed:
Add Ubuntu to tests

jsf9k committed:
Add prepare playbook to update the apt cache for Ubuntu Xenial

For some reason the Xenial Docker image we are using requires this.

jsf9k committed:
Remove line that is apparently unnecessary

jsf9k reviewed (approved):
Thanks for the PR, @monoflo!  This is a nice addition!

I made a few commits to add Ubuntu to the role metadata and molecule tests.

jsf9k commented:
The Travis CI check won't work because Travis isn't enabled on this forked repo.  Once we have two approvals I'll temporarily turn off that requirement for the repo in order to merge in these changes.

jsf9k commented:
The other check is failing because the job is taking too long to run. I will fix this when I have some free time later today.

jsf9k committed:
Wait ten minutes for new signatures

TravisCI appears to be timing out with a wait of only five minutes.

jsf9k committed:
Remove final task that waits for freshclam to run

freshclam will run at next boot on Debian and Ubuntu, since we have
enabled the service.  In the case of Fedora, we just ran freshclam and
verified the presence of /var/lib/clamav/bytecode.cvd.

A further issue is that my TravisCI builds keep petering out on this
task, even though it runs without a problem on my local machine.  I
think this is the age-old problem we have seen with Travis in the
past; too much network activity stalls the build.

jsf9k committed:
Clarify comments

jsf9k committed:
Remove test for freshclam virus signatures

That file will not be present on Debian or Ubuntu until the next boot.

mcdonnnj reviewed (approved):
This is a nice addition for this role. Nice addition @monoflo and thanks for cleaning this up @jsf9k 👍

felddy commented on tasks/main.yml:
```diff
@@ -56,16 +55,10 @@
         creates: /var/lib/clamav/bytecode.cvd
   when: ansible_facts['distribution'] == 'Fedora'
-
-- name: Wait for new signatures to be downloaded and installed by freshclam
```
I know this is perturbing the build... there are a couple reasons I want this completed before the first boot.
 - The AV report generator will be stating these file to get their age.
 - Not all instances have internet access.  It would be nice if they had some signatures even if they are a few weeks old.

jsf9k replied:
OK, I will revert the commits that removed that stuff.

jsf9k replied:
See 1182af8, 13287eb, and 909c548.

felddy replied:
Thanks @jsf9k

felddy reviewed (changes requested):
Let's talk about the removal of the signature download (see comment).

jsf9k committed:
Revert "Remove test for freshclam virus signatures"

This reverts commit 421ea2ffb21c8551bb53c38c3f0e297987cd64cc.

@felddy pointed out that:
* The AV report generator stats these files to get their age
* Not all instances have internet access

jsf9k committed:
Revert "Clarify comments"

This reverts commit 3214fd5988b8fed258a8ce3355eb5a44f0dcb13c.

@felddy pointed out that:
* The AV report generator stats these files to get their age
* Not all instances have internet access

jsf9k committed:
Revert "Remove final task that waits for freshclam to run"

This reverts commit 1ff4609cb4a48865c94de25db3ad18252d7da3ce.

@felddy pointed out that:
* The AV report generator stats these files to get their age
* Not all instances have internet access

felddy reviewed (approved):
Thanks for the contribution @monoflo.  We really appreciate the extra sets of eyes.

Looks good. 👍

jsf9k commented:
I think I can get the build to pass by splitting it into multiple jobs - one for each platform.  I will be experimenting with that now.

dav3r reviewed (approved):
Good work all - looks good to me! 🦅

jsf9k committed:
Move molecule tests

Modify bandit and molecule configurations accordingly.

I am doing this because I want to create some platfrom-specific
scenarios that reuse the existing tests.  This is because the number
of platforms is ovwewhelming TravisCI for this role.

jsf9k committed:
Create distribution-specific molecule scenarios

Also modify TravisCI configuration to run each of these new scenarions
as a separate job.

The hope is that this will allow TravisCI to run to completion without
timing out.

jsf9k committed:
Correct syntax error in TravisCI configuration
